components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/processor/JsonRestProcessor.java

```diff
@@ -100,7 +101,11 @@ protected void processRequest(Exchange exchange) throws SalesforceException {
 
             case SEARCH:
                 // handle known response type
-                exchange.setProperty(RESPONSE_TYPE, TypeReferences.SEARCH_RESULT_TYPE);
+                if (Double.parseDouble(endpoint.getConfiguration().getApiVersion()) >= 37.0) {
```

Comment:
can we somewhat move the version `37.0` number into a constant?

Reply:
I considered that, but couldn't think of a good name. There's no concept that symbolizes any given version. Open to suggestions.

Reply:
True, another option is, we can create a common function in the `AbstractRestProcessor` that sets response type property for `Json` and `XML`, e.g:
```
protected void setSearchResponseType(Object fallbackType) {
if (Double.parseDouble(endpoint.getConfiguration().getApiVersion()) >= 37.0) {
       exchange.setProperty(RESPONSE_TYPE, SearchResult2.class);
} else {
       exchange.setProperty(RESPONSE_TYPE, fallbackType);
     }
}
```

And then called either `setSearchResponseType(TypeReferences.SEARCH_RESULT_TYPE)` or `setSearchResponseType(SearchResults.class)`. WDYT?

Reply:
Hmm. JsonRestProcessor sets the RESPONSE_TYPE property, whereas XmlRestProcessor sets the RESPONSE_CLASS property. And for  < 37.0, the type is different. I'm struggling to find enough in common to extract.

Reply:
But I'm glad you asked this, because I now realize that JsonRestProcessor should be using RESPONSE_CLASS for SearchResult2.

Reply:
ah sorry didn't realize that, you are right, only the version need to be extracted. Anyway, if you can't find a some meaningful naming, is also fine like this as long not much of duplications happening, just wanted to avoid the magic number ;)

Reply:
I know, it bothers me too. But I think `public static final String API_VERSION_37_0 = "37.0";` would bother me more. It brings nothing to the party, ya know?

Reply:
yeah I know, is fine then